File: src/main/java/infinitespire/powers/DeenergizedPower.java
package infinitespire.powers;

import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import com.megacrit.cardcrawl.localization.PowerStrings;
import com.megacrit.cardcrawl.powers.AbstractPower;
import infinitespire.InfiniteSpire;

public class DeenergizedPower extends AbstractPower {

	public static final String powerID = InfiniteSpire.createID("DeenergizedPower");
	private static final PowerStrings strings = CardCrawlGame.languagePack.getPowerStrings(powerID);

	public DeenergizedPower(AbstractPlayer owner, int amount) {
		this.name = strings.NAME;
		this.ID = powerID;
		this.owner = owner;
		this.amount = amount;
		this.updateDescription();
		this.img = InfiniteSpire.Textures.getPowerTexture("energydown.png");
		this.type = PowerType.DEBUFF;
	}

	@Override
	public void updateDescription() {
		this.description = strings.DESCRIPTIONS[0] + amount + strings.DESCRIPTIONS[1];
	}

	@Override
	public void onEnergyRecharge() {
		this.flash();
		AbstractDungeon.player.loseEnergy(this.amount);
	}
}

File: src/main/kotlin/com/blanktheevil/infinitespire/powers/DeEnergizedPower.kt
package com.blanktheevil.infinitespire.powers

import com.blanktheevil.infinitespire.extensions.makeID
import com.blanktheevil.infinitespire.extensions.player
import com.blanktheevil.infinitespire.powers.utils.PowerBuilder
import com.blanktheevil.infinitespire.textures.Textures
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.powers.AbstractPower

class DeEnergizedPower(owner: AbstractPlayer, amount: Int) : Power(
  owner,
  amount,
  BUILDER
) {
  companion object {
    val powerID = "DeenergizedPower".makeID()
    private val BUILDER = PowerBuilder(powerID)
      .debuff()
      .img("energydown.png")
  }

  init {
    name = strings.NAME
    ID = powerID
    this.owner = owner
    this.amount = amount
    this.updateDescription()
    this.img = Textures.powers.get("energydown.png")
    this.type = PowerType.DEBUFF
  }

  override fun updateDesc() {
    this.description = "${strings.DESCRIPTIONS[0]}$amount${strings.DESCRIPTIONS[1]}"
  }

  override fun onEnergyRecharge() {
    this.flash()
    player.loseEnergy(amount)
  }

  override fun makeCopy(): AbstractPower = DeEnergizedPower(this.owner as AbstractPlayer, this.amount)
}